### DISCUSSION TITLE
How to do a correct MongoDB schema

### DISCUSSION THREAD

**Discussion by EverStarck** [prisma/prisma#16967] [Q&A] [answered]
I have the next mongoDB document:

```
"BBVA": 20.2,
"CITI": 30.1,
"total":  50.3,
"year": {
  "2022": {
    "January": {
      "BBVA": 10.2,
      "CITI": 10.1,
      "total":  20.3,
     },
     "February": {
      "BBVA": 10.2,
      "CITI": 10.1,
      "total":  20.3,
     },
     ...
   },
   "2023": {
     "January: {...},
     ...
    }
 }
```

And this is my Prisma schema:

```
type Year {
    2022 Months
    2023 Months
    2024 Months
}

type Months {
    January   TotalMonth
    February  TotalMonth
    March     TotalMonth
    April     TotalMonth
    May       TotalMonth
    June      TotalMonth
    July      TotalMonth
    August    TotalMonth
    September TotalMonth
    October   TotalMonth
    November  TotalMonth
    December  TotalMonth
}

type TotalMonth {
    BBVA  Float
    CITI  Float
    total Float
}

model Total {
    id        String   @id @default(auto()) @map("_id") @db.ObjectId
    BBVA      Float
    CITI      Float
    total     Float
    year      Year
    createdAt DateTime @default(now())
    updatedAt DateTime @updatedAt
}
```

When I try to check my document:

```
  const documentId = "639513143fc45d85ce2e44e0";

  const document = await prisma.total.findUnique({
    where: { id: documentId },
  });
  ```
 I get the next error: 
 `const document = await prisma.total.findUnique(\nError parsing GraphQL query: query parse error: Parse error at 15:7\nUnexpected `2022[IntValue]`\nExpected `Name`\n"}`
  
  
What I'm doing wrong?

**Comment by nurul3101** [comment, marked as answer]
Hello @EverStarck 👋 
Are you using GraphQL with Prisma? Is this error coming from your GraphQL resolver?
The error message indicates it expected a `Name` field, but I can't see a `Name` field in your schema? Have you invoked `npx prisma db push` after making changes to your schema file?

**Reply by EverStarck** [reply]
Hi

No, I'm not using GraphQL. I thought Prisma uses underhood lol

I hadn't run `prisma db push` I just ran `prisma generate`. But I now I did and I get the same error.
This is from `prisma studio`:
```
Message: Error in Prisma Client request: 


Invalid `prisma.total.findMany()` invocation:


Error parsing GraphQL query: query parse error: Parse error at 16:7
Unexpected `2022[IntValue]`
Expected `Name`

  
Query:
{
  "modelName": "Total",
  "operation": "findMany",
  "args": {
    "take": 100,
    "skip": 0,
    "select": {
      "id": true,
      "BBVA": true,
      "CITI": true,
      "total": true,
      "year": true,
      "createdAt": true,
      "updatedAt": true
    }
  }
}
```



btw, here is what `prisma db push` says:
```
Environment variables loaded from .env
Prisma schema loaded from prisma/schema.prisma
Datasource "db"

The database is already in sync with the Prisma schema.

✔ Generated Prisma Client (4.7.1 | library) to ./node_modules/@prisma/client in 86ms
```

**Reply by EverStarck** [reply]
Ooooh I got it.
The problem was I had:
```
...
"year": {
  "2022": {...},
  "2023": {...},
 ```
 
 Apparently, it cannot start with number. I changed from `2022` to `y2022` and so on. Now it works well!
 
 Anyways, thanks for the help @nurul3101

**Reply by nurul3101** [reply]
Oh Yes, that is correct, we have a related [GitHub Issue](https://github.com/prisma/prisma/issues/11390#issuecomment-1028928078) which discusses the same. Glad to hear that you were able to fix this. 👍

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
